src/main/java/org/cbioportal/persistence/mybatisclickhouse/StudyViewMapper.java

```diff
@@ -19,15 +19,14 @@
 import org.cbioportal.web.parameter.GenomicDataFilter;
 
 import java.util.List;
-import java.util.Map;
 
 
 public interface StudyViewMapper {
     List<Sample> getFilteredSamples(@Param("studyViewFilterHelper") StudyViewFilterHelper studyViewFilterHelper);
 
     List<GenomicDataCount> getMolecularProfileSampleCounts(@Param("studyViewFilterHelper") StudyViewFilterHelper studyViewFilterHelper);
     
-    List<AlterationCountByGene> getMutatedGenes(StudyViewFilterHelper studyViewFilterHelper, AlterationFilterHelper alterationFilterHelper);
+    List<AlterationCountByGene> getMutatedGenes(StudyViewFilterHelper studyViewFilterHelper, AlterationFilterHelper alterationFilterHelper, String specificHugoGeneSymbol);
```

Comment:
If you just overload the Repository layer.. you could just keep this the same except pass a list or null then you could just reuse the sql as well

Reply:
Think I got it. Thanks!